src/components/EmojiPicker/EmojiPickerMenu/BaseEmojiPickerMenu.tsx

```diff
@@ -1,78 +1,56 @@
 import {FlashList} from '@shopify/flash-list';
-import PropTypes from 'prop-types';
+import type {ListRenderItem} from '@shopify/flash-list';
 import React, {useMemo} from 'react';
+import type {LegacyRef} from 'react';
 import {StyleSheet, View} from 'react-native';
+import type {StyleProp, ViewStyle} from 'react-native';
+import type {Emoji, HeaderEmoji, PickerEmoji, PickerEmojis} from '@assets/emojis/types';
 import CategoryShortcutBar from '@components/EmojiPicker/CategoryShortcutBar';
 import EmojiSkinToneList from '@components/EmojiPicker/EmojiSkinToneList';
-import refPropTypes from '@components/refPropTypes';
 import Text from '@components/Text';
 import useLocalize from '@hooks/useLocalize';
 import useThemeStyles from '@hooks/useThemeStyles';
 import useWindowDimensions from '@hooks/useWindowDimensions';
-import stylePropTypes from '@styles/stylePropTypes';
 import CONST from '@src/CONST';
+import type {OnyxValue} from '@src/ONYXKEYS';
+import type {EmojiPropTypes, RenderItemProps} from './types';
 
-const emojiPropTypes = {
-    /** The code of the item */
-    code: PropTypes.string.isRequired,
-
-    /** Whether the item is a header or not */
-    header: PropTypes.bool,
-
-    /** Whether the item is a spacer or not */
-    spacer: PropTypes.bool,
-
-    /** Types of an emoji - e.g. different skin types */
-    types: PropTypes.arrayOf(PropTypes.string),
-};
-
-const propTypes = {
+type BaseEmojiPickerMenuProps = {
     /** Indicates if the emoji list is filtered or not */
-    isFiltered: PropTypes.bool.isRequired,
+    isFiltered: boolean;
 
     /** Array of header emojis */
-    headerEmojis: PropTypes.arrayOf(PropTypes.shape(emojiPropTypes)).isRequired,
+    headerEmojis: EmojiPropTypes[];
 
     /** Function to scroll to a specific header in the emoji list */
-    scrollToHeader: PropTypes.func.isRequired,
+    scrollToHeader: (headerIndex: number) => void;
 
     /** Style to be applied to the list wrapper */
-    listWrapperStyle: stylePropTypes,
-
-    /** Reference to the emoji list */
-    forwardedRef: refPropTypes,
+    listWrapperStyle?: StyleProp<ViewStyle>;
 
     /** The data for the emoji list */
-    data: PropTypes.arrayOf(PropTypes.shape(emojiPropTypes)).isRequired,
+    data: PickerEmoji[];
 
     /** Function to render each item in the list */
-    renderItem: PropTypes.func.isRequired,
+    renderItem: ({item, target}: RenderItemProps) => void;
```

Comment:
```suggestion
    renderItem: (item: RenderItemProps) => void;
```

OR can we use ListRenderItem from react-native?

Reply:
> Let me check it 👍

Any luck with that?

> I don't understand why target is removed in the suggestion you made.

I don't see a point in destructuring in a type, that's why I suggested to use just a name for the variable instead of destructured values